web/app/tournament/tourny.py

```diff
@@ -169,3 +172,93 @@ def select_player(tournament_id, tournament_player_id):
         # flash("Could not delete player: {{ tp.name }}")
         return redirect(url_for('.players_index',
             tournament_id=tournament_id))
+
+
+@tournament.route('/<tournament_id>/matches/', methods=['GET', 'POST'])
+def matches_index(tournament_id):
+    if request.method == "GET":
+        matches = Match.query.filter_by(tournament_id=tournament_id)
+        return render_template('tournament_match_index.html',
+        tournament_id=tournament_id,
+        matches=matches)
+
+    if request.method == 'POST':
+        players = TournamentPlayer.query.filter_by(tournament_id=tournament_id)
+        players_list = list(players)
+        shuffle(players_list)
+        while len(players_list) > 0:
+            if len(players_list) % 2 == 0:
+                match = Match(player_1_id = players_list.pop().id,
+                              player_2_id = players_list.pop().id,
+                              tournament_id = tournament_id)
+                db.session.add(match)
+                db.session.commit()
+                match = Match.query.all()[-1]
```

Comment:
As noted above - if you copy parts of the players DB table into the matches DB table, the matches DB table could go out of date if there's a name change. Instead, if you need to present the player's name, you can create a foreign key relationship in SQLalchemy, and do something like {{match.player_1.name}}. There's a bunch of examples of how to set up a foreign key relationships in the existing models.py file.